lib/common/schemas.c

```diff
@@ -1302,3 +1302,32 @@ pcmk__schema_contents(const char *name)
     free(path);
     return contents;
 }
+
+/*!
+ * \internal
+ * \brief Return a list of all schema versions later than the given one
+ *
+ * \param[in] name  The schema version to compare against (for example,
+ *                  "pacemaker-3.1")
+ *
+ * \note The caller is responsible for freeing both the returned list and
+ *       the elements of the list
+ */
+GList *
+pcmk__schemas_later_than(const char *name)
+{
+    GList *lst = NULL;
+    schema_version_t ver;
+
+    if (!version_from_filename(name, &ver)) {
+        return lst;
+    }
+
+    for (int lpc = xml_latest_schema_index(); lpc > 0; lpc--) {
+        if (schema_sort(ver, known_schemas[lpc].version) == -1) {
+            lst = g_list_prepend(lst, strdup(known_schemas[lpc].name));
```

Comment:
Fixed.  Note that there's hundreds of spots we are calling strdup() without checking.

Reply:
Yep we're gradually fixing that as we come across them. At some point it may be worth a project just for that. Aside from a controlled exit, the main benefit is to minimize static analysis complaints